frontend/src/components/common/StudentDetailCommon.vue

```diff
@@ -602,22 +652,69 @@ export default {
         this.genderHint
       ];
     },
-    validateMincode() {
+    validateMinCode() {
+      let message = [];
       if (this.studentCopy) {
         if (!this.studentCopy.mincode) {
           this.mincodeError = false;
-          return [];
+          this.schoolLabel = '';
+          return message;
         } else {
           if (this.studentCopy.mincode.match('^[0-9]\\d*$') && this.studentCopy.mincode.length === 8) {
-            this.mincodeError = false;
-            return [];
+            if (this.isMinCodeChanged) {
+              this.getSchoolName(this.studentCopy.mincode);
+            }
+            if (!this.schoolLabel) {
+              this.mincodeError = true;
+              message.push(this.mincodeHint);
+            } else {
+              this.mincodeError = false;
+            }
+            return message;
           }
         }
       }
+
       this.mincodeError = true;
-      return [
-        this.mincodeHint
-      ];
+      this.schoolLabel = '';
+      message.push(this.mincodeHint);
+      return message;
+    },
+    async getSchoolName(mincode) {
+      try {
+        this.schoolLabel = await this.retrieveSchoolEntity(mincode);
+        if (!this.schoolLabel) {
+          this.mincodeError = true;
+        } else {
+          this.mincodError = false;
+        }
+      } catch (e) {
+        this.schoolLabel = '';
+        this.mincodeError = true;
+      } finally {
+        this.isMinCodeChanged = false;
+      }
+    },
+    retrieveSchoolEntity(mincode) {
+      const params = {
+        params: {
+          mincode,
+        }
+      };
+      if (mincode) {
```

Comment:
looks like the api will be called only when mincode is valid, so not sure if the mincode needs to be checked again.

Reply:
if mincode exists, then it will call api.  so valid or not, it will be called.   just in case if user deletes mincode, then will be skipped.